src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs

```diff
@@ -344,20 +344,11 @@ public void Login()
         public bool OpenRepository()
         {
             var old = Repositories.FirstOrDefault(x => x.Equals(Holder.ActiveRepo));
-            // open the solution selection dialog when the user wants to switch to a different repo
-            // since there's no other way of changing the source control context in VS
             if (!Equals(SelectedRepository, old))
             {
-                if (ErrorHandler.Succeeded(ServiceProvider.GetSolution().OpenSolutionViaDlg(SelectedRepository.LocalPath, 1)))
-                {
-                    ServiceProvider.TryGetService<ITeamExplorer>()?.NavigateToPage(new Guid(TeamExplorerPageIds.Home), null);
-                    return true;
-                }
-                else
-                {
-                    SelectedRepository = old;
-                    return false;
-                }
+                Services.OpenRepository(SelectedRepository.LocalPath);
```

Comment:
If this command blows up, it will take down Visual Studio.
I still need to add exception logging!
Could someone point me in the right direction?